chia/cmds/check_wallet_db.py

```diff
@@ -0,0 +1,303 @@
+            setattr(self, field, value)
+
+    def __repr__(self) -> str:
+        s = ""
+        for f in self.fields:
+            s += f"{f}={getattr(self, f)} "
+        return s
+
+
+class DerivationPath(FromDB):
+    derivation_index: int
+    pubkey: str
+    puzzle_hash: str
+    wallet_type: int
+    wallet_id: int
+    used: int
+    hardened: int
+
+
+class Wallet(FromDB):
+    id: int
+    name: str
+    wallet_type: int
+    data: str
+
+
+def dp_by_wallet_id(derivation_paths: List[DerivationPath]) -> Dict[int, List[int]]:
+    d = defaultdict(list)
+    for dp in derivation_paths:
+        d[dp.wallet_id].append(dp.derivation_index)
+    for k, v in d.items():
+        d[k] = sorted(v)
+    return d
+
+
+def print_min_max_derivation_for_wallets(derivation_paths: List[DerivationPath]) -> None:
+    d = dp_by_wallet_id(derivation_paths)
+    print("Min, Max, Count of derivations for each wallet:")
+    for wallet_id, derivation_index_list in d.items():
+        print(
+            f"Wallet ID {wallet_id:2} derivation index min: {derivation_index_list[0]} "
+            f"max: {derivation_index_list[-1]} count: {len(derivation_index_list)}"
+        )
+
+
+class WalletDBReader:
+    db_wrapper: DBWrapper2  # TODO: Remove db_wrapper member
+    config = {"db_readers": 1}
+    sql_log_path = None
+    verbose = False
+
+    async def get_all_wallets(self) -> List[Wallet]:
+        wallet_fields = ["id", "name", "wallet_type", "data"]
+        async with self.db_wrapper.reader_no_transaction() as reader:
+            cursor = await reader.execute(f"""SELECT {", ".join(wallet_fields)} FROM users_wallets""")
+            rows = await cursor.fetchall()
+            return [Wallet(r, wallet_fields) for r in rows]
+
+    async def get_derivation_paths(self) -> List[DerivationPath]:
+        fields = ["derivation_index", "pubkey", "puzzle_hash", "wallet_type", "wallet_id", "used", "hardened"]
+        async with self.db_wrapper.reader_no_transaction() as reader:
+            cursor = await reader.execute(f"""SELECT {", ".join(fields)} FROM derivation_paths;""")
+            rows = await cursor.fetchall()
+            return [DerivationPath(row, fields) for row in rows]
+
+    async def show_tables(self) -> List[str]:
+        async with self.db_wrapper.reader_no_transaction() as reader:
+            cursor = await reader.execute("""SELECT name FROM sqlite_master WHERE type='table';""")
+            print("\nWallet DB Tables:")
+            print(*([r[0] for r in await cursor.fetchall()]), sep=",\n")
+            print("\nWallet Schema:")
+            print(*(await (await cursor.execute("PRAGMA table_info('users_wallets')")).fetchall()), sep=",\n")
+            print("\nDerivationPath Schema:")
+            print(*(await (await cursor.execute("PRAGMA table_info('derivation_paths')")).fetchall()), sep=",\n")
+            print()
+            return []
+
+    async def check_wallets(self) -> List[str]:
```

Comment:
`check_wallets` could be better - I  could finish moving all the SQL calls up front.